src/lib.rs

```diff
@@ -1,6 +1,349 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
+extern crate byteorder;
+extern crate memmap;
+extern crate rand;
+extern crate regex;
+extern crate time;
+#[macro_use] extern crate bitflags;
+#[macro_use] extern crate lazy_static;
+#[cfg(unix)] extern crate nix;
+#[cfg(windows)] extern crate kernel32;
+
+use byteorder::{LittleEndian, WriteBytesExt};
+use memmap::{Mmap, Protection};
+use rand::Rng;
+use regex::Regex;
+use std::env;
+use std::ffi::{CString, OsString};
+use std::fs;
+use std::fs::{File, OpenOptions};
+use std::io;
+use std::io::{BufReader, Cursor};
+use std::io::prelude::*;
+use std::path::{MAIN_SEPARATOR, Path, PathBuf};
+
+const HDR_LEN: usize = 40;
+const TOC_BLOCK_LEN: usize = 16;
+const CLUSTER_ID_BIT_LEN: usize = 12;
+
+static PCP_TMP_DIR_KEY: &'static str = "PCP_TMP_DIR";
+static MMV_DIR_SUFFIX: &'static str = "mmv";
+
+type Endian = LittleEndian;
+
+#[cfg(unix)]
+fn get_process_id() -> i32 {
+    nix::unistd::getpid()
+}
+
+#[cfg(windows)]
+fn get_process_id() -> i32 {
+    unsafe { kernel32::GetCurrentProcessId() as i32 }
+}
+
+fn get_pcp_root() -> PathBuf {
+    match env::var_os("PCP_DIR") {
+        Some(val) => PathBuf::from(val),
+        None => PathBuf::from(MAIN_SEPARATOR.to_string())
+    }
+}
+
+fn init_pcp_conf(pcp_root: &PathBuf) -> io::Result<()> {
```

Comment:
It doesn't matter much here since this is private code, but note that it's more idiomatic to use `&Path` references for parameters, if not generic `P: AsRef<Path>`.  You can get an owned buffer from there with `to_path_buf()` if really needed, or `join()` will create a new owned `PathBuf` from a shared base `&Path`.